core/store/src/db.rs

```diff
@@ -473,6 +473,18 @@ fn set_compression_options(opts: &mut Options) {
     opts.set_bottommost_zstd_max_train_bytes(max_train_bytes, true);
 }
 
+fn ensure_max_open_files_limit(max_open_files: i32) -> () {
+    // We’re configuring each RocksDB to use max_open_files file descriptors. On top of that we can
+    // have some other file descriptors opened by neard process so we use the value of
+    // 2 * max_open_files to be sure that the binary can correctly run.
+    let (soft, hard) = rlimit::Resource::NOFILE.get().unwrap();
+    let required = 2 * max_open_files as u64;
```

Comment:
Why 1025? That’s a weird number. I understand 1024. I understand 1000, but 1025 is weird.

Anyway, regarding recompress_storage the issue is that it currently assumes max_open_files is 512 so this change will break it. We need https://github.com/near/nearcore/pull/6612 to go in first.

Reply:
Doing some tests I've noticed that with max_open_files=512 the maximum number of simultaneously opened files were 513. That's why 1025. Better safe than sorry :)

Reply:
Well, I fully expect the maximum number of opened file descriptors to be closer to 560 (stdin, stdout and stderr are three, then there’s listening TCP socket, peer connections are each a file descriptor). That’s why we need `max_open_files + some_margin` but the margin is independent of how many descriptors database uses so that’s why I’d choose something round like 1000 or 1024. It doesn’t really matter in the end though.